src/utilities/backendValidation.ts

```diff
@@ -0,0 +1,70 @@
+import { isArray, isPlainObject, extend } from 'lodash';
+
+import { varToLabel } from '@mighty-justice/utils';
+
+// Takes an API response and converts it to a string to string map
+function getFieldErrors (errors: { [key: string]: any }, prefix = '') {
+  const messages: { [key: string]: string } = {};
+  Object.keys(errors).forEach(fieldName => {
+    const fieldKey = [prefix, fieldName].filter(s => !!s).join('-');
+    let fieldErrors = errors[fieldName];
+
+    // If an array, use only the first element
+    if (isArray(fieldErrors)) {
+      fieldErrors = fieldErrors[0];
+    }
+
+    // If an object, recurse
+    if (isPlainObject(fieldErrors)) {
+      extend(messages, getFieldErrors(fieldErrors, fieldKey));
+    }
+
+    // If a simple string, you have your error
+    else {
+      messages[fieldKey] = fieldErrors;
+    }
+  });
+  return messages;
+}
+
+export default function backendValidation (fieldNames: string[], response: object) {
```

Comment:
This seems complicated and very hard to read, let's meet for a few minutes to over this code to see if I can propose something to make it easier to read.

Reply:
It's very complicated and hard to read. I used some very strict Test Driven Development when I originally wrote this file, then commented to my best ability. Open to suggestions though.